### ISSUE TITLE
Stack overflow with inductive having an implicit argument with the same name.

### ISSUE THREAD

**Issue by rlepigre** [coq/coq#11585]
#### Description of the problem

Just try and check the following inductive.
```coq
Inductive type {type : Type} : Type := T : type.
```
**Note**: the fact that the `type` argument is implicit is important.

#### Coq Version

```
The Coq Proof Assistant, version 8.10.2 (February 2020)
compiled on Feb 5 2020 15:10:48 with OCaml 4.07.1
```
@Janno checked with Coq master and the problem is still there.

**Comment by ejgallego** [comment]
Here is the loop:
```
Called from file "interp/constrintern.ml", line 2372, characters 12-48
Called from file "interp/constrintern.ml", line 2345, characters 12-37
```
cc @herbelin

**Comment by herbelin** [comment]
@rlepigre: joli !

I don't see at once how to fix it quickly...

(Hoping you're doing well in passing!)

**Comment by maximedenes** [comment]
Incidentally, this is fixed by #11579

**Comment by herbelin** [comment]
> Incidentally, this is fixed by #11579

Ah, great!

**Comment by rlepigre** [comment]
> @rlepigre: joli !

My initial definition was a lot bigger, and I was really surprised that I ended up with such a small example!

> I don't see at once how to fix it quickly...
> 
> (Hoping you're doing well in passing!)

I'm doing very well, thanks Hugo.

> > Incidentally, this is fixed by #11579
> 
> Ah, great!

Cool! It is always nice when a bug disappears on its own.
